cpp/oneapi/dal/algo/decision_forest/common.cpp

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/decision_forest/common.hpp"
+#include "oneapi/dal/algo/decision_forest/detail/model_impl.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+class detail::descriptor_impl : public base {
+public:
+    double observations_per_tree_fraction      = 1.0;
+    double impurity_threshold                  = 0.0;
+    double min_weight_fraction_in_leaf_node    = 0.0;
+    double min_impurity_decrease_in_split_node = 0.0;
+
+    std::int64_t class_count                    = 1;
+    std::int64_t tree_count                     = 100;
+    std::int64_t features_per_node              = 0;
+    std::int64_t max_tree_depth                 = 0;
+    std::int64_t min_observations_in_leaf_node  = 0;
+    std::int64_t min_observations_in_split_node = 2;
+    std::int64_t max_leaf_nodes                 = 0;
```

Comment:
Is it true that with all the default parameters, a random forest will work?

Reply:
all these values (except of class_count) were taken from daal

Reply:
ok, please check what works with default parameters.

Reply:
checked, that it works with default params.